x/params/subspace/subspace_test.go

```diff
@@ -0,0 +1,203 @@
+package subspace_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params/subspace"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+)
+
+type SubspaceTestSuite struct {
+	suite.Suite
+
+	cdc *codec.Codec
+	ctx sdk.Context
+	ss  subspace.Subspace
+}
+
+func (suite *SubspaceTestSuite) SetupTest() {
+	cdc := codec.New()
+	db := dbm.NewMemDB()
+
+	ms := store.NewCommitMultiStore(db)
+	ms.MountStoreWithDB(key, sdk.StoreTypeIAVL, db)
+	ms.MountStoreWithDB(tkey, sdk.StoreTypeTransient, db)
+	suite.NoError(ms.LoadLatestVersion())
+
+	ss := subspace.NewSubspace(cdc, key, tkey, "testsubspace")
+
+	suite.cdc = cdc
+	suite.ctx = sdk.NewContext(ms, abci.Header{}, false, log.NewNopLogger())
+	suite.ss = ss.WithKeyTable(paramKeyTable())
+}
+
+func (suite *SubspaceTestSuite) TestKeyTable() {
+	suite.True(suite.ss.HasKeyTable())
```

Comment:
nit: `suite` methods only provide assertions. To use require you need to add the following:

```suggestion
	suite.Require().True(suite.ss.HasKeyTable())
```

Ditto with all the `suite` tests. Check the [suite API docs](https://godoc.org/github.com/stretchr/testify/suite#Suite.Require)

Reply:
alternatively, you can use `require.True(suite.T(),  suite.ss.HasKeyTable())`